invoke.go

```diff
@@ -22,16 +22,41 @@ package dig
 
 import (
 	"fmt"
-	"reflect"
-
 	"go.uber.org/dig/internal/digreflect"
 	"go.uber.org/dig/internal/graph"
+	"reflect"
 )
 
-// An InvokeOption modifies the default behavior of Invoke. It's included for
-// future functionality; currently, there are no concrete implementations.
+// An InvokeOption modifies the default behavior of Invoke.
 type InvokeOption interface {
-	unimplemented()
+	applyInvokeOption(*invokeOptions)
+}
+
+type invokeOptions struct {
+	Info *InvokeInfo
+}
+
+// InvokeInfo provides information about an Invoke.
+type InvokeInfo struct {
+	ReqTypes []*Output
+}
+
+// FillInvokeInfo is an InvokeOption that writes info on the types requested
+// by the Invoke function into the specified InvokeInfo.
+func FillInvokeInfo(info *InvokeInfo) InvokeOption {
```

Comment:
It would help to include a sample usage here or an example test elsewhere:

```go
// For example:
//
//    var info dig.InvokeInfo
//    err := c.Invoke(..., dig.FillInvokeInfo(&info))
```

Reply:
Done